libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -56,6 +56,10 @@
 #elif defined(_MSC_VER)
 #  define _CCCL_COMPILER_MSVC     _CCCL_COMPILER_MAKE_VERSION(_MSC_VER / 100, _MSC_VER % 100)
 #  define _CCCL_COMPILER_MSVC2017 (_CCCL_COMPILER_MSVC < _CCCL_COMPILER_MAKE_VERSION(19, 20))
+#  if _CCCL_COMPILER_MSVC2017 && !defined(CCCL_SUPPRESS_MSVC2017_DEPRECATION_WARNING)
+#    warning \
+      "Support for the Visual Studio 2017 (MSC_VER < 1920) is deprecated and will eventually be removed. Define CCCL_SUPPRESS_MSVC2017_DEPRECATION_WARNING to suppress this warning"
+#  endif // CCCL_SUPPRESS_ICC_DEPRECATION_WARNING
```

Comment:
`warning` does not work, we need  
```suggestion
#  if _CCCL_COMPILER_MSVC2017 && !defined(CCCL_SUPPRESS_MSVC2017_DEPRECATION_WARNING)
#    pragma message(\
      "Support for the Visual Studio 2017 (MSC_VER < 1920) is deprecated and will eventually be removed. Define CCCL_SUPPRESS_MSVC2017_DEPRECATION_WARNING to suppress this warning")
#  endif // CCCL_SUPPRESS_ICC_DEPRECATION_WARNING
```

Reply:
Right. I copied this from the ICC deprecation PR. `#warning` did not work on MSVC until C++23 standardized it.